lib/decidim/git_backport_manager.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "English"
+
+module Decidim
+  class GitBackportManager
+    def initialize(pull_request_id:, release_branch:, backport_branch:, working_dir: Dir.pwd)
+      @pull_request_id = pull_request_id
+      @release_branch = release_branch
+      @backport_branch = backport_branch
+      @working_dir = working_dir
+    end
+
+    def call
+      Dir.chdir(working_dir) do
+        exit_if_unstaged_changes
+        self.class.checkout_develop
+        sha_commit = sha_commit_to_backport
+        create_backport_branch!
+        cherrypick_commit!(sha_commit)
+        push_backport_branch!
+      end
+    end
+
+    def self.checkout_develop
+      `git checkout develop`
```

Comment:
I had a failure here because I had uncommitted changes in the backport script because of which it could not checkout `develop` and it aborted the process while the backporter still continued normally.

I suggest we also check it here that it was successful:
```suggestion
      `git checkout develop`
      exit_with_errors("Could not checkout the develop branch. Please mane sure you don't have any uncommitted changes in the current branch.") unless $CHILD_STATUS.exitstatus.zero?
```

This would not likely be a problem once this is merged but better check it to avoid invalid PRs, such as #9494. What it did there is that it failed here but the script proceeded to submit the PR which was then a backport for this PR.